src/main/java/dk/cngroup/lentils/controller/ProgressController.java

```diff
@@ -112,6 +112,16 @@ public ResponseEntity viewHintsForCypherByTeam(final @RequestParam("teamId") Lon
         }
     }
 
+    @GetMapping("/revertHint")
+    public String revertHint(final @RequestParam("teamId") Long teamId,
+                             final @RequestParam("hintId") Long hintId,
+                             final @RequestParam("cypherId") Long cypherId) {
+        Team team = teamService.getTeam(teamId);
+        Hint hint = hintService.getHint(hintId);
+        hintTakenService.revertHint(team, hint);
+        return "redirect:viewHints/" + cypherId + "?teamId=" + teamId;
```

Comment:
Tady bych uz pouzil `String.format()` (Pekny navod treba zde: https://www.javatpoint.com/java-string-format)
Vysledek by pak mohl byt nejaky takovy:
```java
String redirectUrl = String.format( "redirect:viewHints/%d?teamId=%d", cypherId, teamId);
return redirectUrl;

```

Reply:
A jooo, přemýšlel jsem jak to v Javě napsat. Jestli i Spring nemá nějaký URL helper.

Reply:
Redirect uplne odstranen, zavola se reload v javascriptu.